xbmc/cores/paplayer/AudioDecoder.cpp

```diff
@@ -58,7 +58,10 @@ void CAudioDecoder::Destroy()
   m_pcmBuffer.Destroy();
 
   if ( m_codec )
+  {
+    m_codec->DeInit();
```

Comment:
ah ok, we are talking about an addon issue here. I would drop DeInit from the interface ICodec and make the addon CAudioDecoder behave like VideoPlayerCodec. Do deinit in the destructor: https://github.com/xbmc/xbmc/blob/master/xbmc/addons/AudioDecoder.h#L59